packages/block-library/src/image/image-size-picker.native.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Picker } from '@wordpress/components';
+
+const IMAGE_SIZE_THUMBNAIL = 'thumbnail';
+const IMAGE_SIZE_MEDIUM = 'medium';
+const IMAGE_SIZE_LARGE = 'large';
+const IMAGE_SIZE_FULL_SIZE = 'full';
+const DEFAULT_SIZE_SLUG = IMAGE_SIZE_LARGE;
+const sizeOptionLabels = {
+	[ IMAGE_SIZE_THUMBNAIL ]: __( 'Thumbnail' ),
+	[ IMAGE_SIZE_MEDIUM ]: __( 'Medium' ),
+	[ IMAGE_SIZE_LARGE ]: __( 'Large' ),
+	[ IMAGE_SIZE_FULL_SIZE ]: __( 'Full Size' ),
+};
+
+class ImageSizePicker extends Component {
+	getSizeOptionsItems() {
+		return [
+			{ value: IMAGE_SIZE_THUMBNAIL, label: sizeOptionLabels[ IMAGE_SIZE_THUMBNAIL ] },
+			{ value: IMAGE_SIZE_MEDIUM, label: sizeOptionLabels[ IMAGE_SIZE_MEDIUM ] },
+			{ value: IMAGE_SIZE_LARGE, label: sizeOptionLabels[ IMAGE_SIZE_LARGE ] },
+			{ value: IMAGE_SIZE_FULL_SIZE, label: sizeOptionLabels[ IMAGE_SIZE_FULL_SIZE ] },
+		];
+	}
+
+	render() {
+		let picker;
+
+		const sizeOptions = this.getSizeOptionsItems();
+
+		const onPickerPresent = () => {
+			picker.presentPicker();
+		};
+
+		const getImageSizeOptions = () => (
+			<Picker
+				hideCancelButton={ true }
+				title={ __( 'Image Size' ) }
+				ref={ ( instance ) => picker = instance }
+				options={ sizeOptions }
```

Comment:
we could use `map( sizeOptionLabels, ( option, label ) => ( { value: option, label } ) )` instead here

Reply:
Thanks for the tip. Added here: 717d27be68f3073720792815a3f6ae0418812aee
 I didn't realize you could rely on object key orderings, great to know 👍  https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/